app/models/order_detail.rb

```diff
@@ -771,6 +747,34 @@ def problem_order?
     problem
   end
 
+  def can_dispute?
+    in_review? && dispute_at.blank?
+  end
+
+  def in_dispute?
+    dispute_at.present? && dispute_resolved_at.nil? && !canceled?
+  end
+
+  def in_review?
+    reviewed_at.try(:future?) && !in_dispute?
+  end
+
+  def reviewed?
+    reviewed_at.try(:past?) && !in_dispute?
+  end
```

Comment:
An order cannot be `in_review?` and `reviewed?` at the same time. It seemed clearer to me to make them independent and parallel as opposed to calling each other and then also needing additional redundant calls.

I think `past?` might have been confusing on its own, but when paired with `future?`, I thought it made it more obvious than a `< Time.current` and ` > Time.current` And what about `🎁 ?`?